OsmAnd/src/net/osmand/plus/GPXUtilities.java

```diff
@@ -1522,6 +1532,34 @@ private static String readText(XmlPullParser parser, String key) throws XmlPullP
 		return text;
 	}
 
+
+	private static List<ExtensionReadResult> readExtensionsText(XmlPullParser parser, String key,
```

Comment:
Remove readExtensionsText and modify readText. Read and return Map or List of tag/value in readText. The readText will need new name.